lib/ssm_config.rb

```diff
@@ -1,48 +1,59 @@
+require './lib/ssm_storage/ssm_storage_db.rb'
+require './lib/ssm_storage/ssm_storage_yml.rb'
+require './lib/ssm_storage/ssm_storage_empty.rb'
+require 'active_support/core_ext/hash/indifferent_access'
+require 'active_support/time'
+
 class SsmConfig
-  VERSION = '0.1.1'.freeze
-  CONFIG_PATH = 'config'.freeze
-  TABLE_NAME = 'ssm_config_records'.freeze
-  ACTIVE_RECORD_MODEL = 'SsmConfigRecord'.freeze
+  VERSION = '1.0.0'.freeze
+  REFRESH_TIME = (30.minutes).freeze
+
   class << self
     def method_missing(meth, *args, &block)
-      config_file = Rails.root.join(CONFIG_PATH, "#{meth}.yml")
-
-      if ActiveRecord::Base.connection.table_exists? TABLE_NAME
-        if defined? ACTIVE_RECORD_MODEL.constantize and ACTIVE_RECORD_MODEL.constantize.exists?(:file => meth.to_s)
-          return {}
-        else
-          super
-        end
-      elsif File.exist?(config_file)
-        write_config_accessor_for(meth)
-        send(meth)
-      else
-        super
-      end
+      result = populate(meth)
+      super if result == false
+      result
     end
 
     def respond_to_missing?(meth, *_args)
-      config_file = Rails.root.join(CONFIG_PATH, "#{meth}.yml")
-      File.exist?(config_file)
+      config_file = Rails.root.join(SsmStorage::Yml::CONFIG_PATH, "#{meth}.yml")
+      (ActiveRecord::Base.connection.table_exists? SsmStorage::Db::TABLE_NAME) || File.exist?(config_file)
```

Comment:
Sorry for this. Maybe there's a way to move these into methods on the `SsmStorage` classes? Could maybe even recuse the `determine_query` logic here.